runtime/jsTest/src/kotlinx/serialization/DynamicSerializerTest.kt

```diff
@@ -0,0 +1,191 @@
+package kotlinx.serialization
+
+import kotlinx.serialization.internal.IntArraySerializer
+import kotlinx.serialization.internal.IntSerializer
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonConfiguration
+import kotlinx.serialization.modules.EmptyModule
+import kotlinx.serialization.modules.getContextualOrDefault
+import kotlin.test.*
+
+
+class DynamicSerializerTest {
+    @Serializable
+    data class Data(val a: Int)
+
+    @Serializable
+    open class DataWrapper(open val s: String, val d: Data?)
+
+    @Serializable
+    data class NestedList(val a: String, val list: List<Int>)
+
+    @Serializable
+    data class ListOfLists(val l: List<List<Data>>)
+
+    @Serializable
+    data class MapWrapper(val m: Map<String, Int>)
+
+    @Serializable
+    data class ComplexMapWrapper(val m: Map<String, Data>)
+
+    @Serializable
+    data class WithChar(val a: Char)
+
+    @Serializable
+    data class AllTypes(
+        val b: Byte,
+        val s: Short,
+        val i: Int,
+        val f: Float,
+        val d: Double,
+        val c: Char,
+        val B: Boolean,
+        val S: String
+    )
+
+    @Serializable
+    data class EnumWrapper(val e: Color)
+
+    private inline fun <reified T : Any> assertDynamicForm(
+        data: T,
+        serializer: SerializationStrategy<T>? = null,
+        noinline assertions: ((T, dynamic) -> Unit)? = null
```

Comment:
Should `assertions` be allowed to be null? It seems like there are currently quite a few test cases that don't assert anything regarding the input/output, but that is likely important?